userspace/libsinsp/test/public_sinsp_API/events_set.cpp

```diff
@@ -137,7 +137,47 @@ TEST(events_set, names_to_event_set)
 	event_names.insert("syncfs");
 	event_codes = libsinsp::events::names_to_event_set(event_names);
 	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_E));
-	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_X));
-	ASSERT_EQ(event_codes.size(), 6); // enter/exit events for each event name
+	ASSERT_TRUE(event_codes.contains(PPME_GENERIC_X));ASSERT_EQ(event_codes.size(), 6); // enter/exit events for each event name
+}
+
+// Tests that no generic ppm sc is mapped to an event too
+// basically, avoid that someone added a new event mapping a once-generic syscall,
+// and forgot to update libscap/linux/scap_ppm_sc.c::g_events_to_sc_map.
+TEST(events_set, generic_no_events)
+{
+	auto generic_ev_set = libsinsp::events::set<ppm_event_code>({PPME_GENERIC_E, PPME_GENERIC_X});
+	auto generic_sc_set = libsinsp::events::event_set_to_sc_set(generic_ev_set);
+	auto final_ev_set = libsinsp::events::sc_set_to_event_set(generic_sc_set);
+	ASSERT_EQ(final_ev_set, generic_ev_set);
+}
+
+TEST(events_set, event_set_to_sc_set)
+{
+	auto all_sc = libsinsp::events::all_sc_set();
+	auto all_events = libsinsp::events::all_event_set();
+	auto events_to_sc = libsinsp::events::event_set_to_sc_set;
+	auto sc_to_events = libsinsp::events::sc_set_to_event_set;
+
+	libsinsp::events::set<ppm_event_code> all_syscalls_tracepoints_events;
+	for (auto ev_code : all_events)
+	{
+		if (libsinsp::events::is_tracepoint_event(ev_code) ||
+		   libsinsp::events::is_syscall_event(ev_code))
+		{
+			all_syscalls_tracepoints_events.insert(ev_code);
+		}
+	}
+
+	ASSERT_EQ(all_syscalls_tracepoints_events, sc_to_events(all_sc));
+	ASSERT_EQ(all_sc, events_to_sc(all_syscalls_tracepoints_events));
+	ASSERT_EQ(all_sc, events_to_sc(sc_to_events(all_sc)));
+	ASSERT_EQ(all_syscalls_tracepoints_events, sc_to_events(events_to_sc(all_syscalls_tracepoints_events)));
+
+	auto sc_to_names = libsinsp::events::sc_set_to_names;
+	auto names_to_sc = libsinsp::events::names_to_sc_set;
+	ASSERT_EQ(all_sc, names_to_sc(sc_to_names(all_sc)));
 
+	auto events_to_names = libsinsp::events::event_set_to_names;
```

Comment:
@FedeDP `event_set_to_names` not yet fixed in this PR, same issues as in https://github.com/falcosecurity/libs/pull/915, I used those tests for testing the 889 branch.

```
Expected equality of these values:
  event_names.size()
    Which is: 3
  event_names2.size()
    Which is: 239
```

```
TEST(events_set, event_set_to_names)
{
	auto event_names = std::unordered_set<std::string>{"openat2","execveat","syncfs"};
	auto event_codes = libsinsp::events::names_to_event_set(event_names);
	auto event_names2 = libsinsp::events::event_set_to_names(event_codes);
	ASSERT_EQ(event_names.size(), event_names2.size()); // This fails!
}

```